crates/biome_js_analyze/src/assist/source/organize_imports.rs

```diff
@@ -1,144 +1,521 @@
+    /// Provides a code action to sort the imports and exports in the file using a built-in or custom order.
+    ///
+    /// Imports and exports are first separated into chunks, before being sorted.
+    /// Imports or exports of a chunk are then grouped according to the user-defined groups.
+    /// Within a group, imports are sorted using a built-in order that depends on the import/export kind, whether the import/export has attributes and the source being imported from.
+    ///
+    /// ```js,ignore
+    /// import A from "@my/lib" with { "attribute1": "value" };
+    /// ^^^^^^^^       ^^^^^^^         ^^^^^^^^^^^^^^^^^^^^^
+    ///   kind         source                attributes
+    ///
+    /// export * from "@my/lib" with { "attribute1": "value" };
+    /// ^^^^^^^^       ^^^^^^^         ^^^^^^^^^^^^^^^^^^^^^
+    ///   kind         source                attributes
+    /// ```
+    ///
+    /// Note that **source** is also often called **specifier** in the JavaScript ecosystem.
+    ///
+    ///
+    /// ## Chunk of imports and chunks of exports
+    ///
+    /// A **chunk** is a sequence of adjacent imports or exports.
+    /// A chunk contains only imports or exports, not both at the same time.
+    /// The following example includes two chunks.
+    /// The first chunk consists of the three imports and the second chunk consists of the three exports.
+    /// The first `export` that appears ends the first chunk and starts a new one.
+    ///
+    /// ```js,ignore
+    /// import A from "a";
+    /// import * as B from "b";
+    /// import { C } from "c";
+    /// export * from "d";
+    /// export * as F from "e";
+    /// export { F } from "f";
+    /// ```
+    ///
+    /// Chunks also ends as soon as a statement or a **side-effect import** (also called _bare import_) is encountered.
+    /// Each side-effect import consists in a chunk:
+    /// The following example contains six chunks:
+    ///
+    /// 1. The first chunk contains the two first `import` and ends with the appearance of the first side-effect import `import "x"`.
+    /// 2. The second chunk contains only the side-effect import `import "x"`.
+    /// 3. The third chunk contains only the side-effect import `import "y"`.
+    /// 4. The fourth chunk contains a single `import`; The first `export` ends it.
+    /// 5. The fifth chunk contains the first `export`; The function declaration ends it.
+    /// 6. The sixth chunk contains the last two `export`.
+    ///
+    /// ```js,ignore
+    /// // chunk 1
+    /// import A from "a";
+    /// import * as B from "b";
+    /// // chunk 2
+    /// import "x";
+    /// // chunk 3
+    /// import "y";
+    /// // chunk 4
+    /// import { C } from "c";
+    /// // chunk 5
+    /// export * from "d";
+    /// function f() {}
+    /// // chunk 6
+    /// export * as E from "e";
+    /// export { F } from "f";
+    /// ```
```

Comment:
I would suggest moving the explanations below the example. That way, the example code actually follows the line "The following example contains six chunks:", and you give the example as context before doing the explaining.

```suggestion
    /// ```js,ignore
    /// // chunk 1
    /// import A from "a";
    /// import * as B from "b";
    /// // chunk 2
    /// import "x";
    /// // chunk 3
    /// import "y";
    /// // chunk 4
    /// import { C } from "c";
    /// // chunk 5
    /// export * from "d";
    /// function f() {}
    /// // chunk 6
    /// export * as E from "e";
    /// export { F } from "f";
    /// ```
    ///
    /// 1. The first chunk contains the two first `import` and ends with the appearance of the first side-effect import `import "x"`.
    /// 2. The second chunk contains only the side-effect import `import "x"`.
    /// 3. The third chunk contains only the side-effect import `import "y"`.
    /// 4. The fourth chunk contains a single `import`; The first `export` ends it.
    /// 5. The fifth chunk contains the first `export`; The function declaration ends it.
    /// 6. The sixth chunk contains the last two `export`.
```